demo.java

```diff
@@ -0,0 +1,166 @@
+     * - Once the status is marked as DELIVERED, we can stop tracking the package  
+     *  
+     * A Sample files looks as below: 
+     *  FILE-NAME-> '8787323232232332--55322798-dd29-4a04-97f4-93e18feed554' 
+     *   >status:IN TRANSIT 
+     *   >timestamp: 1573410202 
+     *   >Other fields like...tracking history and address 
+     */ 
+    public String handleRequest(ScheduledEvent scheduledEvent, Context context) { 
+ 
+        final LambdaLogger logger = context.getLogger(); 
+        try { 
+            processShipmentUpdates(logger); 
+            return "SUCCESS"; 
+        } catch (final Exception ex) { 
+            logger.log(String.format("Failed to process shipment Updates in %s due to %s", scheduledEvent.getAccount(), ex.getMessage())); 
+            throw new RuntimeException(ex); 
+        } 
+    } 
+ 
+ 
+    private void processShipmentUpdates(final LambdaLogger logger) throws InterruptedException { 
+ 
+        final List<String> bucketsToProcess = Constants.BUCKETS_TO_PROCESS; 
+        final ConcurrentHashMap<String, Pair<Long, String>> latestStatusForTrackingNumber = new ConcurrentHashMap<String, Pair<Long, String>>(); 
+        final ConcurrentHashMap<String, List<KeyVersion>> filesToDelete = new ConcurrentHashMap<String, List<DeleteObjectsRequest.KeyVersion>>(); 
+        bucketsToProcess.parallelStream().forEach(bucketName -> { 
+            final List<KeyVersion> filesProcessed = processEventsInBucket(bucketName, logger, latestStatusForTrackingNumber); 
+            filesToDelete.put(bucketName, filesProcessed); 
+    }); 
+         
+        final AmazonS3 s3Client = EventHandler.getS3Client(); 
+        //Create a new file in the Constants.SUMMARY_BUCKET 
+        logger.log("Map of statuses -> " + latestStatusForTrackingNumber); 
+        String summaryUpdateName = Long.toString(System.currentTimeMillis()); 
+         
+        EventHandler.getS3Client().putObject(Constants.SUMMARY_BUCKET, summaryUpdateName, latestStatusForTrackingNumber.toString());
+     
+        Waiter waiter = EventHandler.getS3Client().waiters().objectExists();
+        try{
+            waiter.run(new WaiterParameters<>(new GetObjectMetadataRequest(Constants.SUMMARY_BUCKET, summaryUpdateName)));
+            deleteProcessedFiles(filesToDelete);
+            logger.log("All updates successfully processed");
+        } catch (WaiterTimedOutException e){
+            throw new RuntimeException("Failed to write sumary status, will be retried in 15 minutes");
+        }
+    } 
+ 
+    private List<KeyVersion> processEventsInBucket(String bucketName, LambdaLogger logger, ConcurrentHashMap<String, Pair<Long, String>> latestStatusForTrackingNumber) { 
+ 
+        final AmazonS3 s3Client = EventHandler.getS3Client(); 
+        logger.log("Processing Bucket: " + bucketName); 
+ 
+        ObjectListing files = s3Client.listObjects(bucketName); 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.